### PR TITLE
Create project payer deploy tx and add it to v2 tools modal

### PR DESCRIPTION
**Pull request by johnnyd-eth**
## What does this PR do and why?

As title.

Addressed all the comments you made here @aeolianeth :

https://github.com/jbx-protocol/juice-interface/pull/898

Note: the process here is quite different. Moved most text into it's own modal and made another modal for when the address is deployed to copy it. 

## Screenshots or screen recordings

https://user-images.githubusercontent.com/96150256/166171027-1600a92f-cc2e-4bfb-bfd2-7054fe59041d.mp4

## Acceptance checklist

- [ ] I have evaluated the [Approval Guidelines](https://github.com/jbx-protocol/juice-interface/blob/main/CONTRIBUTING.md#approval-guidelines) for this PR.
- [ ] I have tested this PR in [all supported browsers](https://github.com/jbx-protocol/juice-interface/blob/main/CONTRIBUTING.md#supported-browsers).

### CODE REVIEW

File: src/components/v2/V2Project/LaunchProjectPayerModal.tsx [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+        <CopyTextButton value={projectPayerAddress} style={{ fontSize: 25 }} />
+        <p style={{ marginTop: 30 }}>
+          <Trans>
+            This address will disappear when you close this window.{' '}
+            <strong>Copy the address and save it now</strong>.
+          </Trans>
+        </p>
+        <p>
+          <Trans>
+            If you lose your address please contact the Juicebox team through{' '}
```
Comment (wraeth-eth) [inline src/components/v2/V2Project/LaunchProjectPayerModal.tsx:143]: grammar: `If you lose your address, please contact the Juicebox team...`

File: src/components/v2/V2Project/LaunchProjectPayerModal.tsx [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+            entity like the Coinbase app,{' '}
+            <strong>
+              tokens can't be issued to their personal wallets and will be lost
+            </strong>
+            .
+          </Trans>
+        </p>
+        <p>
+          <Trans>
+            This deploys a contract to the blockchain, and will incur
```
Comment (wraeth-eth) [inline src/components/v2/V2Project/LaunchProjectPayerModal.tsx:114]: nit: `This deploys a contract to the blockchain and will incur significant gas fees.` drop the `,`

**Review by wraeth-eth — APPROVED** [review]
Couple of small comments but otherwise LGTM

File: src/components/v2/V2Project/LaunchProjectPayerModal.tsx (hunk 2 above)
Comment (johnnyd-eth) [reply src/components/v2/V2Project/LaunchProjectPayerModal.tsx:114]: 100%

**Comment by johnnyd-eth** [comment]
Thanks @wraeth-eth and @aeolianeth, addressed y'alls latest

File: src/components/v2/V2Project/LaunchProjectPayerModal.tsx [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+            entity like the Coinbase app,{' '}
+            <strong>
+              tokens can't be issued to their personal wallets and will be lost
+            </strong>
+            .
+          </Trans>
+        </p>
+        <p>
+          <Trans>and will incur significant gas fees.</Trans>
+        </p>
```
Comment (aeolianeth) [inline src/components/v2/V2Project/LaunchProjectPayerModal.tsx:114]: **question** should this be here?

File: src/components/v2/V2Project/LaunchProjectPayerModal.tsx [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+      </TransactionModal>
+      <Modal
+        visible={confirmedModalVisible}
+        onOk={() => setConfirmedModalVisible(false)}
+        cancelButtonProps={{ hidden: true }}
```
Comment (aeolianeth) [inline src/components/v2/V2Project/LaunchProjectPayerModal.tsx:119]: TIL, nice

File: src/components/v2/V2Project/LaunchProjectPayerModal.tsx [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+          const txHash = result?.transaction?.hash
+          if (!txHash) {
+            return
+          }
+
+          const txReceipt = await readProvider.getTransactionReceipt(txHash)
+          const newProjectPayerAddress =
+            getProjectPayerAddressFromReceipt(txReceipt)
+          if (newProjectPayerAddress === undefined) {
+            return
```
Comment (aeolianeth) [inline src/components/v2/V2Project/LaunchProjectPayerModal.tsx:71]: is this an error case? should we display an error notification to the user if so?

**Review by aeolianeth — CHANGES_REQUESTED** [review]

File: src/components/v2/V2Project/LaunchProjectPayerModal.tsx (hunk 3 above)
Comment (johnnyd-eth) [reply src/components/v2/V2Project/LaunchProjectPayerModal.tsx:114]: lol not sure how got there, thanks

File: src/components/v2/V2Project/LaunchProjectPayerModal.tsx (hunk 5 above)
Comment (johnnyd-eth) [reply src/components/v2/V2Project/LaunchProjectPayerModal.tsx:71]: Yeah good call. Done

**Review by aeolianeth — APPROVED** [review]
Thanks @johnnyd-eth tested and works great! This feature will help drive MAP. Merging

**Pull request merged**